hera_mc/geo_handling.py

```diff
@@ -158,68 +153,60 @@ def get_station_types(self, add_stations=True):
         else:
             station_present = False
         return station_present
 
-    def is_in_connections(self, station_name, active_date=None, return_antrev=True):
+    def find_antenna_at_station(self, station, query_date):
         """
-        checks to see if the station_name is in the connections database
-            (which means it is also in parts)
-        if active_date is None, it will return True/False if ever connected
-        if active_date is given, it will check whether it was connected at that time
+        checks to see what antenna is at a station
 
-        return True/False unless (1) active_date is provided and (2) return_antrev is True
+        Returns None or the active antenna_name and revision (must be an active
+        antenna_name for the query_date)
 
         Parameters:
         ------------
-        station_name:  string name of station
-        active_date:  astropy Time to check if active, default is None
-        return_antrev:  boolean flag to return True/False or antrev tuple, default is True
+        station:  station name as string.
+        query_date:  is the astropy Time for contemporary antenna
         """
 
-        active_date = cm_utils._get_astropytime(active_date)
-        ustn = station_name.upper()
+        query_date = cm_utils._get_astropytime(query_date)
+        ustn = station.upper()
         connected_antenna = self.session.query(part_connect.Connections).filter(
-            func.upper(part_connect.Connections.upstream_part) == ustn)
-        if connected_antenna.count() > 0:
-            antenna_connected = True
-        else:
-            antenna_connected = False
-        if antenna_connected and active_date is not None:
-            counter = 0
-            for connection in connected_antenna.all():
-                connection.gps2Time()
-                if cm_utils._is_active(active_date, connection.start_date,
-                                       connection.stop_date):
-                    if return_antrev:
-                        antenna_connected = (connection.downstream_part,
-                                             connection.down_part_rev)
-                    else:
-                        antenna_connected = True
-                    counter += 1
-                else:
-                    antenna_connected = False
-            if counter > 1:
-                s = "More than one active connection for {}".format(station_name)
-                warning.warn(s)
-                antenna_connected = False
-        return antenna_connected
+            (func.upper(part_connect.Connections.upstream_part) == ustn) &
+            (query_date.gps >= part_connect.Connections.start_gpstime))
+        ctr = 0
+        for conn in connected_antenna:
+            if conn.stop_gpstime is None or query_date.gps <= conn.stop_gpstime:
+                antenna_connected = copy.copy(conn)
+                ctr += 1
+        if ctr == 0:
+            antenna_connected = None
```

Comment:
If the `ctr == 0` codepath is triggered, the code will crash, because `None` does not have fields called `downstream_part` and `down_part_rev`. If I understand the code correctly I think you should just `return None` here.